### PR TITLE
feat: Add `no-install` option

### PR DESCRIPTION
**Pull request by jaw0r3k**
**Please describe the changes this PR makes and why it should be merged:**

Whilst it is better to install packages automatically, we can provide an option for the user to disable it.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/create-discord-bot/template/TypeScript/src/events/index.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import type { ClientEvents } from 'discord.js';
 
-export interface Event<T extends keyof ClientEvents = keyof ClientEvents> {
+export type Event<T extends keyof ClientEvents = keyof ClientEvents> = {
 	execute(...parameters: ClientEvents[T]): Promise<void> | void;
 	name: T;
 	once?: boolean;
-}
+};
```
Comment (jaw0r3k) [inline packages/create-discord-bot/template/TypeScript/src/events/index.ts:7]: Souji's pr extends that

File: packages/create-discord-bot/template/TypeScript/src/events/index.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import type { ClientEvents } from 'discord.js';
 
-export interface Event<T extends keyof ClientEvents = keyof ClientEvents> {
+export type Event<T extends keyof ClientEvents = keyof ClientEvents> = {
```
Comment (vladfrangu) [inline packages/create-discord-bot/template/TypeScript/src/events/index.ts:3]: Why was this changed again?

Comment (vladfrangu) [reply packages/create-discord-bot/template/TypeScript/src/events/index.ts:3]: I see it was changed in https://github.com/discordjs/discord.js/pull/9570 too, so ignore this. That said, changing it here and in that PR will probably cause merge conflicts

**Review by vladfrangu — APPROVED** [review]

**Review by sdanialraza — APPROVED** [review]

**Comment by Jiralite** [comment]
This needs a rebase.

File: packages/create-discord-bot/template/TypeScript/src/events/index.ts (hunk 1 above)
Comment (Jiralite) [reply packages/create-discord-bot/template/TypeScript/src/events/index.ts:7]: Do you mean #9570?

File: packages/create-discord-bot/template/TypeScript/src/events/ready.ts [hunk 3]
```diff
@@ -7,4 +7,4 @@ export default {
 	async execute(client) {
 		console.log(`Ready! Logged in as ${client.user.tag}`);
 	},
-} satisfies Event<'ready'>;
+} satisfies Event<Events.ClientReady>;
```
Comment (Jiralite) [inline packages/create-discord-bot/template/TypeScript/src/events/ready.ts:10]: This seems like an unrelated change.

**Review by SpaceEEC — APPROVED** [review]

File: packages/create-discord-bot/bin/index.ts [hunk 4]
```diff
@@ -40,6 +40,7 @@ program
 	})
 	.option('--typescript', 'Whether to use the TypeScript template.')
 	.option('--javascript', 'Whether to use the JavaScript template.')
+	.option('--noInstall', 'Whether to not automatically install the packages.')
```
Comment (Jiralite) [inline packages/create-discord-bot/bin/index.ts:43]: Should this not be `--no-install`?

**Review by Jiralite — CHANGES_REQUESTED** [review]

File: packages/create-discord-bot/bin/index.ts (hunk 4 above)
Comment (jaw0r3k) [reply packages/create-discord-bot/bin/index.ts:43]: Crawl used diffrent syntax so for continuity i also used it

Comment (Jiralite) [reply packages/create-discord-bot/bin/index.ts:43]: This (and that) should change.

**Review by Jiralite — APPROVED** [review]
